spec/controllers/users_controller_spec.rb

```diff
@@ -75,4 +124,29 @@
       let(:action) { get :show, id: luke.id }
     end
   end
+
+  describe "GET new_with_invitation_token" do
+    it "renders the :new view template" do
+      invitation = Fabricate(:invitation)
+      get :new_with_invitation_token, token: invitation.token
+      expect(response).to render_template :new
+    end
+
+    it "sets @user with recipient's email" do
+      invitation = Fabricate(:invitation)
+      get :new_with_invitation_token, token: invitation.token
+      expect(assigns(:user).email).to eq(invitation.recipient_email)
+    end
+
+    it "sets @invitation_token" do
+      invitation = Fabricate(:invitation)
+      get :new_with_invitation_token, token: invitation.token
+      expect(assigns(:invitation_token)).to eq(invitation.token)
+    end
+
+    it "redirects to exipred token page for invalid tokens" do
+      get :new_with_invitation_token, token: 'random_blah'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.